### DISCUSSION TITLE
Use keyword versions of `target_link_libraries`

### DISCUSSION THREAD

**Discussion by JAicewizard** [duckdb/duckdb#12811] [Ideas]
Using the keyword versions would provide extensions more freedom over how to link their libraries. This can be useful in some cases

Pros:
- This will give the freedom to extensions to chose whether or not the dependencies are linked privately or publicly.

Cons:
- This would require updating most extensions

Although I thought this usecase would be more useful, it ends up being only useful in a handful of situations. Although I still think this would be a good change, the benefits do not seem to outway the cons.

**Comment by samansmink** [comment]
Thanks for opening a discussion @JAicewizard.

The non-keyword version of `target_link_libraries` has been slightly annoying a few times, but I agree that currently it might not be worth it to break all extension builds right now.

If more reasons pop up, we can add them to this discussion to reconsider!
